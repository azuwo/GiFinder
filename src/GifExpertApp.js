import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories = ["useless"] }) => {
  //const categories = ['One Punch', 'Samurai X','Dragon Ball']
  //const [categories, setCategories] = useState(['One Punch'])

  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>Gif Finder No One Will Ever Use</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => {
          return <GifGrid key={category} category={category} />;
        })}
      </ol>
    </>
  );
};

export default GifExpertApp;
