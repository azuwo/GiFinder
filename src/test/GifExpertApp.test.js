import React from 'react'
import {shallow} from 'enzyme'
import '@testing-library/jest-dom'
import GifExpertApp from '../GifExpertApp'

describe('Test in GifExpertApp', () => {
    test('should return component', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot()
    })

    test('should show category list', () => {
        const categories = ['One Punch','Dragon Ball']
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories}/>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
})