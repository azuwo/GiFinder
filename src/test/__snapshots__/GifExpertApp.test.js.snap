// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpertApp should return component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test in GifExpertApp should show category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
